sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -665,6 +667,11 @@ void release(String bundleDescriptorId, BundleProcessor bundleProcessor) {
       }
     }
 
+    /** Discard an active {@link BundleProcessor} instead of being re-used. */
+    void discard(BundleProcessor bundleProcessor) {
+      activeBundleProcessors.remove(bundleProcessor.getInstructionId());
```

Comment:
Do we want to do `bundleProcessor.setInstructionId(null);` before discarding it to avoid the same race condition as `reset`?

Reply:
It is unnecessary since reset will not be invoked.